docs/ldml/tr35-keyboards.md

```diff
@@ -632,6 +621,26 @@ Indicates that:
 2.  If a transform is escaped, output the contents of the buffer.
 3.  During a transform, hide the contents of the buffer as the user is typing.
 
+_Attribute:_ `normalization`
+
+> Normalization will not typically be the responsibility of the keyboard author, rather this will be managed by the implementation.
+> Output from the keyboard, following application of all transform rules, will be normalized to implementation or application-requested form.
```

Comment:
This makes it sound like normalization only happens on output. We should be a bit more explicit about what normalization allowed means for when things need to be matched (transforms, display).

Reply:
Yes; the additional normalization details should be an implementation guideline -- that implementers should normalize 'context' inputs from the app and all strings within keyboard to NFD internally. (NFD because it allows consistent interleaving of markers.)

Reply:
PTAL think i addressed this